radix-engine-utils/src/rocks_db_metrics/mod.rs

```diff
@@ -0,0 +1,544 @@
+use radix_engine_store_interface::interface::{
+    CommittableSubstateDatabase, DatabaseUpdates, DbPartitionKey, DbSortKey, DbSubstateValue,
+    PartitionEntry, SubstateDatabase,
+};
+use radix_engine_stores::{
+    memory_db::InMemorySubstateDatabase,
+    rocks_db::{BlockBasedOptions, Options, RocksdbSubstateStore},
+};
+use std::{cell::RefCell, collections::BTreeMap, path::PathBuf, time::Duration};
+
+/// Substate store with read time measurements for RocksDB and In Memory DB.
+pub struct SubstateStoreWithMetrics<S>
+where
+    S: SubstateDatabase + CommittableSubstateDatabase,
+{
+    db: S,
+    pub read_metrics: RefCell<BTreeMap<usize, Vec<Duration>>>,
+}
+
+impl SubstateStoreWithMetrics<RocksdbSubstateStore> {
+    pub fn new_rocksdb(path: PathBuf) -> Self {
+        let mut factory_opts = BlockBasedOptions::default();
+        factory_opts.disable_cache();
+
+        let mut opt = Options::default();
+        opt.set_disable_auto_compactions(true);
```

Comment:
Out of curiosity, why do we disable `auto_compactions`?

Reply:
I just wanted to ensure that database will not be automatically compacted to replicate more real usage scenario, where we won't be using database in compacted state all the time.